Applications/Utils/ModelPreparation/PartitionMesh/PartitionMesh.cpp

```diff
@@ -86,78 +87,78 @@ int main(int argc, char* argv[])
-        mesh_partitioner.writeMETIS(BaseLib::dropFileExtension(ifile_name) +
-                                    ".mesh");
-    }
-    else
+    const int num_partitions = nparts.getValue();
+
+    // Execute mpmetis via system(...)
+    if (num_partitions > 1 && exe_metis_flag.getValue())
     {
-        const int num_partitions = nparts.getValue();
+        INFO("METIS is running ...");
+        const std::string exe_name = argv[0];
+        const std::string exe_path = BaseLib::extractPath(exe_name);
+        INFO("Path to mpmetis is: \n\t%s", exe_path.c_str());
 
-        // Execute mpmetis via system(...)
-        if (num_partitions > 1 && exe_metis_flag.getValue())
-        {
-            INFO("METIS is running ...");
-            const std::string exe_name = argv[0];
-            const std::string exe_path = BaseLib::extractPath(exe_name);
-            INFO("Path to mpmetis is: \n\t%s", exe_path.c_str());
-
-            const std::string mpmetis_com =
-                exe_path + "/mpmetis " + " -gtype=nodal " + file_name_base +
-                ".mesh " + std::to_string(nparts.getValue());
-
-            const int status = system(mpmetis_com.c_str());
-            if (status != 0)
-            {
-                INFO("Failed in system calling.");
-                INFO("Return value of system call %d ", status);
-                return EXIT_FAILURE;
-            }
-        }
-        else if (num_partitions == 1 && exe_metis_flag.getValue())
+        const std::string mpmetis_com =
+            exe_path + "/mpmetis " + " -gtype=nodal " + file_name_base +
+            ".mesh " + std::to_string(nparts.getValue());
+
+        const int status = system(mpmetis_com.c_str());
+        if (status != 0)
         {
-            // The mpmetis tool can not be used for 'partitioning' in only one
-            // domain. For this reason the according files are written for just
-            // one domain in the metis output format in the following.
-            auto writePartitionFile =
-                [&file_name_base](std::string const& file_name_extension,
-                                  std::size_t number) {
-                    std::string const name(file_name_base +
-                                           file_name_extension);
-                    std::ofstream os(name);
-                    if (!os)
-                        OGS_FATAL("Couldn't open file '%s' for writing.",
-                                  name.c_str());
-                    for (std::size_t n(0); n < number; ++n)
-                        os << "0\n";
-                };
-
-            writePartitionFile(".mesh.npart.1", number_of_nodes);
-            writePartitionFile(".mesh.epart.1", number_of_elements);
+            INFO("Failed in system calling.");
+            INFO("Return value of system call %d ", status);
+            return EXIT_FAILURE;
         }
+    }
+    else if (num_partitions == 1 && exe_metis_flag.getValue())
+    {
```

Comment:
@wenqing Please comment.

Reply:
As suggested by @chleh, that branch can be replaced by OGS_FATAL as
```
OGS_FATAL("Partitioning the mesh into one domain is unnecessary because OGS reads vtu mesh data directly when np=1 ")
```